receiver/vcenterreceiver/factory.go

```diff
@@ -0,0 +1,117 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vcenterreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver/internal/metadata"
+)
+
+const (
+	typeStr = "vcenter"
+)
+
+type vcenterReceiverFactory struct {
+	receivers map[*Config]*vcenterReceiver
+}
+
+// NewFactory returns the receiver factory for the vcenterreceiver
+func NewFactory() component.ReceiverFactory {
+	f := &vcenterReceiverFactory{
+		receivers: make(map[*Config]*vcenterReceiver),
+	}
+	return component.NewReceiverFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithMetricsReceiver(f.createMetricsReceiver),
+	)
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		MetricsConfig: &MetricsConfig{
+			ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+				ReceiverSettings:   config.NewReceiverSettings(config.NewComponentID(typeStr)),
+				CollectionInterval: 5 * time.Minute,
+			},
+			TLSClientSetting: configtls.TLSClientSetting{},
+			Metrics:          metadata.DefaultMetricsSettings(),
+			Endpoint:         "",
+			Username:         "",
+			Password:         "",
```

Comment:
```suggestion
```
`""` is the zero-value for strings. Typically don't explicitly initialize fields to their zero-values.